actors.md

```diff
@@ -2,10 +2,94 @@
 
 Any implementations of the Filecoin actors must be exactly byte for byte compatible with the go-filecoin actor implementations. The pseudocode below tries to capture the important logic, but capturing all the detail would require embedding exactly the code from go-filecoin, so for now, its simply informative pseudocode. The algorithms below are correct, and all implementations much match it (including go-filecoin), but details omitted from here should be looked for in the go-filecoin code.
 
-This spec decsribes a set of actors that operate within the [Filecoin State Machine](state-machine.md). All types are defined in [the basic type encoding spec](data-structures.md#basic-type-encodings).
+This spec describes a set of actors that operate within the [Filecoin State Machine](state-machine.md). All types are defined in [the basic type encoding spec](data-structures.md#basic-type-encodings).
 
+
+- [Init Actor](#init-actor)
 - [Storage Market Actor](#storage-market-actor)
 - [Storage Miner Actor](#storage-miner-actor)
+- [Payment Channel Broker Actor](#payment-channel-broker-actor)
+
+
+## Init Actor
+The init actor is responsible for creating new actors on the filecoin network. This is a built-in actor and cannot be replicated. In the future, this actor will be responsible for loading new code into the system (for user programmable actors).
+
+```go
+type InitActor struct {
+    // Mapping from Address to ID, for lookups.
+		AddressMap map[Address]BigInt
+}
+```
+
+### `Exec(code Cid, params []Param) Address`
+
+>  This method is the core of the `Init Actor`. It handles instantiating new actors and assigning them their IDs.
+
+#### Parameters
+
+| Name     | Type       | Description                                                  |
+| -------- | ---------- | ------------------------------------------------------------ |
+| `code`   | `Cid`      | A pointer to the location at which the code of the actor to create is stored. |
+| `params` | `[] Param` | The parameters passed to the constructor of the actor.       |
+
+
+
+`Param` is the type representing any valid arugment that can be passed to a function.
+
+TODO: Find a better place for this definition.
+
+
+```go
+func Exec(code Cid, params []byte) Address {
+    // Get the actor ID for this actor.
+    actorID = len(self.AddressMap)
+
+    // Make sure that only the actors defined in the spec can be launched.
+    if !IsBuiltinActor(code) {
+        Fatal("cannot launch actor instance that is not a builtin actor")
+    }
+
+    // Ensure that singeltons can be only launched once.
+    // TODO: do we want to enforce this? If so how should actors be marked as such?
+    if IsSingletonActor(code) && self.hasInstance(code) {
```

Comment:
tricky.  I see what youre doing here, making it so that we can use the init actor itself to launch the other protected actors at network launch. I think maybe we can fake that all for now, and just have a simple whitelist here.

Reply:
I just realized something, do we have a way to get the code Cid from the actor currently? If not we should have this, such that I can check if a certain actor is spawned from this particular code cid

Reply:
we should definitely have that, i don't think its exposed right now in an easy to use way